compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/SkikoUIView.kt

```diff
@@ -108,43 +119,22 @@ internal class SkikoUIView : UIView, UIKeyInputProtocol, UITextInputProtocol {
         }
     }
 
-    // merging two constructors might cause a binary incompatibility, which will result in a unclear linking error,
-    // if a project using newer compose depends on an older compose transitively
-    // https://youtrack.jetbrains.com/issue/KT-60399
-    @Suppress("UNUSED") // public API
     constructor(
-        skiaLayer: IOSSkiaLayer,
-        frame: CValue<CGRect> = CGRectNull.readValue(),
-        pointInside: (Point, UIEvent?) -> Boolean = { _, _ -> true }
-    ) : this(skiaLayer, frame, pointInside, skikoUITextInputTrains = object :
-        SkikoUITextInputTraits {})
-
-    constructor(
-        skiaLayer: IOSSkiaLayer,
         frame: CValue<CGRect> = CGRectNull.readValue(),
         pointInside: (Point, UIEvent?) -> Boolean = { _, _ -> true },
-        skikoUITextInputTrains: SkikoUITextInputTraits,
-        drawCompletionCallback: () -> Unit = { }
     ) : super(frame) {
-        _skiaLayer = skiaLayer
         _pointInside = pointInside
-        _skikoUITextInputTrains = skikoUITextInputTrains
-
-
-        // TODO: remove weak wrapper if we are not going to implement lifecycle-bound behavior
-        val weakSkiaLayer = WeakReference(skiaLayer)
 
         _redrawer = MetalRedrawer(
             _metalLayer,
-            drawCallback = { surface ->
-                weakSkiaLayer.get()?.draw(surface)
+            drawCallback = {
+                delegate?.draw(it)
```

Comment:
I suggest to use original name and also add a type declaration for better understanding:
```suggestion
            drawCallback = { surface: Surface
                delegate?.draw(surface)
```

Reply:
Sounds good, updated